### ISSUE TITLE
Add additional checks

### ISSUE THREAD

**Issue by WojciechKretowicz** [ModelOriented/DALEX#121]
In explain.R are some not described input errors. If you pass them to explain(), you will get the nondescriptive error:
* Check if `data` is null while weights are not. 
* Check if `data` is not a data frame while `y` is provided.
* Chech if `data` is null while `y` is provided.

**Comment by pbiecek** [comment]
Thanks. 
Good ideas. I am not sure about the second one (not data frame = tibble or data.table)?.

**Comment by WojciechKretowicz** [comment]
You can pass `data` as an object that do not have function `nrow()`. Then if you provided `y` you get the error from this if statement, because `nrow(data)` is `NULL`.

```
if (length(y) != nrow(data)) {
      verbose_cat("  -> target variable   :  length of 'y' is different than number of rows in 'data' (",color_codes$red_start,"WARNING",color_codes$red_end,") \n", verbose = verbose)
    }
```

**Comment by maksymiuks** [comment]
Check that were implemented

* Check if `data` is null while weights are not. 
* Chech if `data` is null while `y` is provided.
* Check if `predict_function` is `function` class object
* Check if `residual_function` is `function` class object
* Check if `label` is `character` class object and conversion if not.

I decided not to implement that proposed check: 
* Check if `data` is not a data frame while `y` is provided.

as it is not necessary for `data` to be data.frame. It can be matrix or tbl in some applications.

https://github.com/ModelOriented/DALEX/pull/140
